registry_provider.go

```diff
@@ -0,0 +1,265 @@
+package tfe
+
+import (
+	"context"
+	"fmt"
+	"net/url"
+)
+
+// Compile-time proof of interface implementation.
+var _ RegistryProviders = (*registryProviders)(nil)
+
+// RegistryProviders describes all the registry provider-related methods that the Terraform
+// Enterprise API supports.
+//
+// TFE API docs: https://www.terraform.io/docs/cloud/api/providers.html
+type RegistryProviders interface {
+	// List all the providers within an organization.
+	List(ctx context.Context, organization string, options *RegistryProviderListOptions) (*RegistryProviderList, error)
+
+	// Create a registry provider.
+	Create(ctx context.Context, organization string, options RegistryProviderCreateOptions) (*RegistryProvider, error)
+
+	// Read a registry provider.
+	Read(ctx context.Context, providerID RegistryProviderID, options *RegistryProviderReadOptions) (*RegistryProvider, error)
+
+	// Delete a registry provider.
+	Delete(ctx context.Context, providerID RegistryProviderID) error
+}
+
+// registryProviders implements RegistryProviders.
+type registryProviders struct {
+	client *Client
+}
+
+// RegistryName represents which registry is being targeted
+type RegistryName string
+
+// List of available registry names
+const (
+	PrivateRegistry RegistryName = "private"
+	PublicRegistry  RegistryName = "public"
+)
+
+// RegistryProviderIncludeOps represents which jsonapi include can be used with registry providers
+type RegistryProviderIncludeOps string
+
+// List of available includes
+const (
+	RegistryProviderVersionsInclude RegistryProviderIncludeOps = "registry-provider-versions"
```

Comment:
I don't see this documented or in the API. Is it possible to include this relationship?

Reply:
🤔 You're right, when I re-reviewed the documentation, it is not clear that this include is an option. I tested it locally with this request:

```
curl --location --request GET 'https://LOCALURL/api/v2/organizations/amazing-org/registry-providers/private/amazing-org/zebra-provider?include=registry-provider-versions' \
--header 'Content-Type: application/vnd.api+json' \
--header 'Authorization: Bearer TOKEN' \
--header 'Cookie: __profilin=p%3Dt' \
--data-raw '{
  "data": {
    "type": "registry-providers",
    "attributes": {
      "name": "test-registry-provider-5de6cb61-b5ea-e72e-13a2-c9aab0b5c3ac",
      "namespace": "amazing-org",
      "registry-name": "private"
    }
  }
}'
```

And the response does include the registry provider versions (see the attached json). I think we'll need to update the docs. What are your thoughts?

[response.txt](https://github.com/hashicorp/go-tfe/files/8757919/response.txt)